core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1466,17 +1531,29 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     }
   }
 
-  def getCheckpointDir = checkpointDir
+  def getCheckpointDir = {
+    assertNotStopped()
+    checkpointDir
+  }
 
   /** Default level of parallelism to use when not given by user (e.g. parallelize and makeRDD). */
-  def defaultParallelism: Int = taskScheduler.defaultParallelism
+  def defaultParallelism: Int = {
```

Comment:
This throws an exception because `taskScheduler` is null:

```
scala> sc.defaultParallelism
java.lang.NullPointerException
	at org.apache.spark.SparkContext.defaultParallelism(SparkContext.scala:1461)
	at $iwC$$iwC$$iwC$$iwC.<init>(<console>:13)
	at $iwC$$iwC$$iwC.<init>(<console>:18)
	at $iwC$$iwC.<init>(<console>:20)
	at $iwC.<init>(<console>:22)
	at <init>(<console>:24)
	at .<init>(<console>:28)
	at .<clinit>(<console>)
	at .<init>(<console>:7)
	at .<clinit>(<console>)
	at $print(<console>)
```